Tests/SwiftParserTest/SyntaxTransformVisitor.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  SyntaxTransformVisitor.swift
+//  
+//
+//  Created by Zoe Carver on 8/25/22.
+//
+
+import XCTest
+import SwiftParser
+import SwiftSyntax
+
+final class SyntaxTransformVisitorTest: XCTestCase {
+  public func testFunctionCounter() {
+    struct FuncCounter: SyntaxTransformVisitor {
+      public func visitAny(_ node: Syntax) -> Int {
+        visitChildren(node).reduce(0, +)
+      }
+ 
+      public func visit(_ node: FunctionDeclSyntax) -> Int {
+        visitChildren(node).reduce(1, +)
+      }
+    }
+    XCTAssertNoThrow(try {
+      let parsed = try Parser.parse(source: """
+          func foo() {
+            public func foo() {
+              func foo() {
+                /*Unknown token */0xG
+              }
+              func bar() {
+                /*Unknown token */0xG
+              }
+            }
+          }
+          """)
+      let count = FuncCounter().visit(parsed)
+      XCTAssertEqual(count, 4)
+    }())
+  }
+  
+  public func testStringify() {
+    struct Stringify: SyntaxTransformVisitor {
```

Comment:
Similarly to the test name, what do you think about renaming this:

```suggestion
    /// Assuming there is a single functioned declaration in the source file, print the type of that function.
    struct PrintFunctionType: SyntaxTransformVisitor {
```